File: benchmarks/closures_deep.py
def make_adder(n):
    def adder(x):
        return x + n
    return adder

fns = []
i = 0
while i < 200:
    fns.append(make_adder(i))
    i = i + 1

total = 0
round_ = 0
while round_ < 500:
    for f in fns:
        total = total + f(round_)
    round_ = round_ + 1
print(total)

File: benchmarks/cpp/closures_deep.cpp
#include <cstdio>
#include <functional>
#include <vector>

std::function<long long(long long)> make_adder(long long n) {
    return [n](long long x) { return x + n; };
}

int main() {
    std::vector<std::function<long long(long long)>> fns;
    for (long long i = 0; i < 200; i++) fns.push_back(make_adder(i));

    long long total = 0;
    for (long long round = 0; round < 500; round++) {
        for (auto& f : fns) total += f(round);
    }
    printf("%lld\n", total);
}
